tree/dataframe/src/RNTupleDS.cxx

```diff
@@ -135,6 +135,7 @@ class RArraySizeField final : public ROOT::Experimental::RFieldBase {
    }
 
 public:
+   static std::string TypeName() { return "std::size_t"; }
```

Comment:
I think here and for `RRDFCardinalityField`, we can try to remove `TypeName()`. This static method is only used for compile-time type names in the `RField<T>` constructor.